src/ios/OpenTokPlugin.m

```diff
@@ -346,20 +346,102 @@ - (void)subscriberDidConnectToStream:(OTSubscriberKit*)sub{
-    [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData];
-    
+    [self triggerJSEvent: @"subscriberEvents" withType: @"connected" withData: eventData];
+    [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData]; // Backwards compatibility
+}
+- (void)subscriberDidDisconnectFromStream:(OTSubscriberKit*)sub{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    NSString* streamId = sub.stream.streamId;
+    [eventData setObject:streamId forKey:@"streamId"];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"disconnected" withData: eventData];
 }
-- (void)subscriber:(OTSubscriber*)subscrib didFailWithError:(OTError*)error{
+- (void)subscriber:(OTSubscriber*)sub didFailWithError:(OTError*)error{
     NSLog(@"subscriber didFailWithError %@", error);
     NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
-    NSString* streamId = subscrib.stream.streamId;
+    NSString* streamId = sub.stream.streamId;
     NSNumber* errorCode = [NSNumber numberWithInt:1600];
     [eventData setObject: errorCode forKey:@"errorCode"];
     [eventData setObject:streamId forKey:@"streamId"];
     [self triggerJSEvent: @"sessionEvents" withType: @"subscribedToStream" withData: eventData];
 }
+- (void)subscriberVideoDataReceived:(OTSubscriber*)sub{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    //[self triggerJSEvent: @"subscriberEvents" withType: @"videoDataReceived" withData: eventData];
+}
+- (void)subscriberVideoDisabled:(OTSubscriberKit*)sub reason:(OTSubscriberVideoEventReason)reason{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    NSString* reasonData = (reason == OTSubscriberVideoEventPublisherPropertyChanged ? @"publishVideo" : @"");
+    reasonData = (reason == OTSubscriberVideoEventSubscriberPropertyChanged ? @"subscribeToVideo" : reasonData);
+    reasonData = (reason == OTSubscriberVideoEventQualityChanged ? @"quality" : reasonData);
+    
+    [eventData setObject: reasonData forKey:@"reason"];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDisabled" withData: eventData];
+}
+- (void)subscriberVideoDisableWarning:(OTSubscriberKit*)sub{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDisableWarning" withData: eventData];
+}
+- (void)subscriberVideoDisableWarningLifted:(OTSubscriberKit*)sub{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    [self triggerJSEvent: @"subscriberEvents" withType: @"videoDisableWarningLifted" withData: eventData];
+}
+- (void)subscriberVideoEnabled:(OTSubscriberKit*)sub reason:(OTSubscriberVideoEventReason)reason{
+    NSMutableDictionary* eventData = [[NSMutableDictionary alloc] init];
+    NSString* reasonData = (reason == OTSubscriberVideoEventPublisherPropertyChanged ? @"publishVideo" : @"");
```

Comment:
why modify the reason? can't we just save it as a string?

If not, I prefer a switch statement or a regular if/elseif/else rather than using the ternary operator and keep reassigning the value. (It's kind of hard to read)

Reply:
The reason why I did this implementation(and it is up for debate) is because of the iOS SDK implementation of the video enabled event [OTSubscriberVideoEventReason](https://tokbox.com/developer/sdks/ios/reference/Constants/OTSubscriberVideoEventReason.html).

According to the documentation it's an integer based enum while in the [Android](https://tokbox.com/developer/sdks/android/reference/com/opentok/android/SubscriberKit.VideoListener.html#onVideoEnabled-com.opentok.android.SubscriberKit-java.lang.String-) SDK it's a string. Same goes for the [browser[(https://tokbox.com/developer/sdks/js/reference/VideoEnabledChangedEvent.html) SDK. 

Unless I have missed something which is indeed possible ;-)

Reply:
No, you are correct. But can we do a switch statement or an else if instead of the multiple ternary operators?

Reply:
I have changed it to a simple ifelse statement